packages/nightingale-sunburst/src/nightingale-sunburst.js

```diff
@@ -0,0 +1,358 @@
+        0,
+        0,
+        segment.data[this["id-attribute"]] ===
+          this.activeSegment?.data?.[this["id-attribute"]]
+          ? undefined
+          : spaceAvailableToDraw - 2
+      );
+      context.restore();
+    }
+  }
+
+  renderActiveSegmentInfo(context) {
+    if (this.activeSegment) {
+      context.textAlign = "left";
+      context.font = "bold 12px Arial";
+      context.fillText("Name:", 10, 10);
+      context.fillText(this["weight-attribute-label"], 10, 40);
+      context.font = "12px Arial";
+      context.fillText(this.activeSegment.data[this["name-attribute"]], 15, 25);
+      context.fillText(
+        getValue(this.activeSegment.data, this["weight-attribute"]),
+        15,
+        55
+      );
+    }
+  }
+
+  selectNodeByPosition(x, y) {
+    const h = Math.sqrt(x * x + y * y);
+    const angle1 = x === 0 ? 0 : Math.atan(y / x);
+    const angle2 = y === 0 ? 0 : Math.atan(x / y);
+    const prev = this.activeSegment;
+    this.activeSegment = null;
+    this.root
+      .descendants()
+      .filter((d) => d.depth)
+      .forEach((d) => {
+        // if is in the right ring
+        let angle = 0;
+        if (d.y1 > h && h > d.y0) {
+          if (x > 0 && y > 0) angle = angle1;
+          else if (x < 0 && y > 0) {
+            angle = Math.PI / 2 + Math.abs(angle2);
+          } else if (x < 0 && y < 0) {
+            angle = Math.PI + Math.abs(angle1);
+          } else if (x > 0 && y < 0) {
+            angle = 1.5 * Math.PI + Math.abs(angle2);
+          }
+          if (d.x1 > angle && angle > d.x0) this.activeSegment = d;
+        }
+      });
+    if (prev !== this.activeSegment) {
+      if (this.activeSegment?.data && !this.activeSegment.data.lineage) {
+        this.activeSegment.data.lineage = getLineageFromNode(
+          this.activeSegment,
+          this["name-attribute"],
+          this["id-attribute"]
+        );
+      }
+      this.dispatchEvent(
+        new CustomEvent("taxon-hover", {
+          detail: this.activeSegment?.data,
+          bubbles: true,
+          cancelable: true,
+        })
+      );
+      this.renderCanvas();
+    }
+  }
+
+  createRenderRoot() {
+    return this;
+  }
+
+  firstUpdated() {
+    this.getElementsByTagName("canvas")?.[0].addEventListener(
+      "click",
+      (event) => {
+        if (!this.root) return;
+        this.holdSegment = !this.holdSegment;
+        if (!this.holdSegment) {
+          this.selectNodeByPosition(
+            event.offsetX - this.side / 2,
+            event.offsetY - this.side / 2
+          );
+        }
+      }
+    );
+    this.getElementsByTagName("canvas")?.[0].addEventListener(
+      "mousemove",
+      (event) => {
+        if (!this.root || (this.activeSegment && this.holdSegment)) return;
+        this.selectNodeByPosition(
+          event.offsetX - this.side / 2,
+          event.offsetY - this.side / 2
+        );
+      }
+    );
+  }
```

Comment:
Would put all of that in a connectedCallback, attaching the event listeners on this, and removing them on disconnectedCallback. It means the event listeners might need to be defined somewhere else (in the constructor?) and used in those 2 different lifecycle callbacks